### DISCUSSION TITLE
Bug in Execute-ProcessAsUser PSAppDeployToolkit 3.9.2

### DISCUSSION THREAD

**Discussion by mortykai** [PSAppDeployToolkit/PSAppDeployToolkit#795] [General]
When called without specifying TempPath throws exception:

Message        : Cannot bind argument to parameter 'LiteralPath' because it is an empty string.
InnerException : 

FullyQualifiedErrorId : ParameterArgumentValidationErrorEmptyStringNotAllowed,Microsoft.PowerShell.Commands.TestPathCommand
ScriptStackTrace      : at Execute-ProcessAsUser<Process>, C:\Windows\IMECache\61b88c36-8d29-4701-b895-3d07273a6d97_25\AppDeployToolkit\AppDeployToolkitMain.ps1: line 7476
                        at <ScriptBlock>, C:\Windows\IMECache\61b88c36-8d29-4701-b895-3d07273a6d97_25\Deploy-Application.ps1: line 262
                        at <ScriptBlock>, <No file>: line 1
                        at <ScriptBlock>, <No file>: line 1

PositionMessage : At C:\Windows\IMECache\61b88c36-8d29-4701-b895-3d07273a6d97_25\AppDeployToolkit\AppDeployToolkitMain.ps1:7476 char:36
                  +         If (Test-Path -LiteralPath $executeAsUserTempPath -PathType ' ...
                  +                                    ~~~~~~~~~~~~~~~~~~~~~~

**Comment by BradEley** [comment]
@mortykai  - It looks like the the check for the tempPath parameter is incorrect.
On line 7433 of _\AppDeployToolkit\AppDeployToolkitMain.ps1_, change `If ($null -eq $tempPath) {` to `If ($tempPath) {`

**Comment by mjr4077au** [comment]
This has all been rewritten for v4.
